baybe/kernels/basic.py

```diff
@@ -43,3 +45,45 @@ class MaternKernel(Kernel):
 
     Only takes the values 0.5, 1.5 or 2.5. Larger values yield smoother interpolations.
     """
+
+    lengthscale_prior: Optional[Prior] = field(default=None)
+    """An optional prior on the kernel lengthscale."""
+
+    lengthscale_initial_value: Optional[float] = field(
```

Comment:
I guess here (and also in `ScaleKernel` we also need to rule out `nan` and `inf`. Have a look at the new `finite_float` validator I added #225 (as well as the corresponding `finite_floats` hypothesis strategy that you can use in the tests)

Reply:
That's not a problem, there is the corresponding `optional` callable provided by attrs – have added the validators accordingly. Also, I think they are necessary because of the following reasons:
1. The an appropriate error needs to be raised by us directly during the attempt of creating an object with invalid input, and not by downstream code (such as gpytorch). First, we have no control of what happens downstream. Second, without the validator, you'd still be able to create a broken instance of the Kernel, and you'll see the (random) error only later when triggering the computation. That causes problems specifically when using it in our API.
2. With the argument "we don't need to protect people" you'd pretty much question ALL our validators :D This one here is no special case.